src/Collection/index.js

```diff
@@ -113,9 +112,7 @@ export default class Collection<Record: Model> {
   }
 
   async _update(record: Record): Promise<void> {
-    record._hasPendingUpdate = false
-    await this.database.adapter.batch([['update', record]])
-    this._onRecordUpdated(record)
+    await this.database.batch(record)
```

Comment:
that's silly ;) This method isn't doing anything now. Change it to use database.batch at the source — in `Model.update`